pyiso/nyiso.py

```diff
@@ -147,19 +147,25 @@ def get_any(self, label, parser, dates_list=None):
             dates_list = self.dates()
 
         # fetch and parse all csvs
-        for date in dates_list:
+        date = min(dates_list)
+        while date in dates_list:
             for csv in self.fetch_csvs(date, label):
-
                 try:
                     pieces.append(parser(csv))
                 except AttributeError:
                     pass
 
+
             # if fetch_csvs cannot get the individual days, it gets the whole month
             # Shortcut the loop if any call to fetch_csvs gets all dates in dates_list
             try:
-                if (pieces[-1].index[-1] - timedelta(days=1)).date() > max(dates_list):
+                if (pieces[-1].index[-1].date() - timedelta(days=1)) > max(dates_list):
+                    print('quit cause i finished')
```

Comment:
Please use `LOGGER` (i.e. `from pyiso import LOGGER`) for appropriate log levels and some more informative message. For example:

`LOGGER.debug(self.NAME + ': Breaking iteration of CSV data. End of date range has been reached.')`